File: psychology-education-frontend/src/API/ClientService.js
import axios from "axios";
import getAuthHeader from "Utils/getAuthHeader";

const API_URL = "http://localhost:8080/api/client/";

export default class ClientService {
    static async getCourses() {
        const response = await axios.get(API_URL + "courses", {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getCourseById(id) {
        const response = await axios.get(API_URL + `courses/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getStudiedTopics(id) {
        const response = await axios.get(API_URL + `courses/studiedTopics/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getTopicById(id) {
        const response = await axios.get(API_URL + `topics/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async deleteFavouriteCourse(id) {
        const response = await axios.delete(API_URL + `courses/favourites/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async addFavouriteCourse(id) {
        const response = await axios.get(API_URL + `courses/favourites/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async addRecord(id) {
        const response = await axios.get(API_URL + `courses/records/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async addProgress(id) {
        const response = await axios.get(API_URL + `courses/progress/${id}`, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async addReview(id, {review}) {
        const response = await axios.put(API_URL + `reviews/${id}`, review, {
            headers: getAuthHeader(),
        });
        return response;
    }

    static async getCertificate(id) {
        const response = await axios.get(API_URL + `courses/certificate/${id}`, {
            headers: getAuthHeader(),
            responseType: 'arraybuffer'
        });
        return response;
    }
}

File: psychology-education-backend/src/main/java/com/psychology/service/ClientService.java
package com.psychology.service;

import com.psychology.dto.ReviewDTO;
import com.psychology.dto.SelectStudiedTopics;
import com.psychology.entity.Record;
import com.psychology.entity.*;
import com.psychology.exception.NotFoundException;
import org.springframework.core.io.Resource;

import java.io.IOException;
import java.util.List;

public interface ClientService {

    Iterable<Course> getCourses();

    Course getCourseById(Long id) throws NotFoundException;

    Topic getTopicById(Long id) throws NotFoundException;

    Favourite addFavouriteCourse(Long id, User user) throws NotFoundException;

    Review addReview(Long id, User user, ReviewDTO reviewDTO) throws NotFoundException;

    Record addRecord(Long id, User user) throws NotFoundException;

    Progress addProgress(Long id, User user) throws NotFoundException;

    Long deleteFavouriteCourse(Long id, User user) throws NotFoundException;

    Resource getCertificate(Long courseId, User user) throws NotFoundException, IOException;

    List<SelectStudiedTopics> getStudiedTopics(Long courseId, User user) throws NotFoundException;
}
